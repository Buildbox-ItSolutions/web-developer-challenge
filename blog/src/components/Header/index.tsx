import { LogoHeader } from "../../utils/img/logoHeader";
import { Logo, LogoContainer } from "./style";

export default function Header() {
    return (
        <LogoContainer>
            <Logo>
                <LogoHeader />
            </Logo>
        </LogoContainer>
    );
}
