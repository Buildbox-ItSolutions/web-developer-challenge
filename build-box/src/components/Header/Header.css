header {
  width: 100%;
  background-color: #2b2b2b;
}