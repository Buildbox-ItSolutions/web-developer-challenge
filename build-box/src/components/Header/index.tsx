import BxLogo from '../../assets/images/bx-logo.svg';
import styled from 'styled-components';

const StylizedHeader = styled.header`
	align-items: center;
	background-color: #2b2b2b;
	display: flex;
	height: 93px;
	justify-content: center;
	position: fixed;
	width: 100vw;
`;

const StylizedLogo = styled.img`
	height: 45px;
	object-fit: contain;
	width: 103px;
`;

function Header() {
	return (
		<StylizedHeader>
			<StylizedLogo
				src={BxLogo}
				className="header"
				alt="buildbox web challenge logo"
			/>
		</StylizedHeader>
	);
}

export default Header;
