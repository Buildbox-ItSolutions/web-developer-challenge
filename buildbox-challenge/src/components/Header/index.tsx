import { Container } from './style'
import buildBoxLogo from '../../assets/buildBoxLogo.svg'
export const Header = () => {
  return (
    <Container>
        <img src = { buildBoxLogo } alt='build-box-logo'/>
    </Container>
  )
}
