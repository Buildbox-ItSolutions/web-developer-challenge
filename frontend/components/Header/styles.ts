import styled from 'styled-components';

export const Header = styled.header`
    width: 100%;
    height: 93px;
    grid-area: header;
    background-color: #2b2b2b;
    display: flex;
    align-items: center;
    justify-content: center;
    position: fixed;
    z-index: 100;
`;

export const Img = styled.img`
    width: 103px;
    height: 45px;
`;
