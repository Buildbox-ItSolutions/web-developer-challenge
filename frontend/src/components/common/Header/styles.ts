// External libraries
import styled from 'styled-components'

// Styled
import theme from '../../../styles/theme'

export const Container = styled.div`
  width: 100%;
  height: 93px;

  display: flex;
  justify-content: center;
  align-items: center;

  background: ${theme.colors.headers_background};

  position: fixed;
  top: 0;
  left: 0;
  right: 0;
`

export const Content = styled.div`
  width: fit-content;
  height: 100%;

  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-direction: column;

  p {
    font-size: 12px;
    font-weight: 100;
    text-transform: uppercase;

    text-align: left;

    line-height: 14px;
    letter-spacing: 0.3px;

    color: ${theme.colors.gray_250};
  }
`
