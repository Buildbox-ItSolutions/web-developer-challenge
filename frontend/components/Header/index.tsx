import { FC } from 'react';

import { Header, Img } from './styles';

const HeaderComponent: FC = () => {
    return (
        <Header>
            <Img src="/bx-logo.png" alt="Logo" data-test="header-logo" />
        </Header>
    );
};

export default HeaderComponent;
