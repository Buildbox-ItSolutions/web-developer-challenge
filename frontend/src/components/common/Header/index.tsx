// External libraries
import React from 'react'

// Styled
import { Container, Content } from './styles'

const Header: React.FC = () => {
  return (
    <Container>
      <Content>
        <h5>buildbox</h5>
        <p>web challenge</p>
      </Content>
    </Container>
  )
}

export default Header
