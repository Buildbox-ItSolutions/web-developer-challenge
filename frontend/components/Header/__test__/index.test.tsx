import 'jest-styled-components';

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';

jest.mock('next/image');

describe('Header', () => {
    it('must render without errors', () => {
        const wrapper = renderer.create(<Header />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('must display the logo ', () => {
        const wrapper = shallow(<Header />);

        expect(wrapper.find(`[data-test='header-logo']`).exists()).toBeTruthy();
    });
});
