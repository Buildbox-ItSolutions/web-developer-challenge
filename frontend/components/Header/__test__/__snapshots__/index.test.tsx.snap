// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header must render without errors 1`] = `
.c0 {
  width: 100%;
  height: 93px;
  grid-area: header;
  background-color: #2b2b2b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  z-index: 100;
}

.c1 {
  width: 103px;
  height: 45px;
}

<header
  className="c0"
>
  <img
    alt="Logo"
    className="c1"
    data-test="header-logo"
    src="/bx-logo.png"
  />
</header>
`;
